in_elasticsearch: Add descriptions for elasticsearch server parameter and default values (#1135)
fluent/fluent-bit-docs: pipeline/inputs/elasticsearch.md

The plugin supports the following configuration parameters:

| Key | Description | Default value |
| :--- | :--- | :--- |
| buffer\_max\_size | Set the maximum size of buffer. | 4M |
| buffer\_chunk\_size | Set the buffer chunk size. | 512K |
| tag\_key | Specify a key name for extracting as a tag. | `NULL` |
| meta\_key | Specify a key name for meta information. | "@meta" |

> | hostname | Specify hostname or FQDN. This parameter is effective for sniffering node information. | "localhost" |

Reviewer: Typo on `sniffering` but I'm not sure what it is supposed to mean here - do we get some extra metadata from it?

Author: Ah, in Elasticsearch/OpenSearch context, sniffer means auto discovery of the server nodes.
ref: https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/current/sniffer.html

Should we use another terms and descriptions for this parameter?

Reviewer: Ah right so it is auto-discovery of Elastic nodes. What does this parameter actually do, what should you set it to?

I think it should be the hostname of an Elastic node right? Then the library uses the `info` endpoint to discover the others? Say you're running in a multi-nic of container environment, what would this be set to?

Author: > I think it should be the hostname of an Elastic node right? Then the library uses the `info` endpoint to discover the others?

Yes, this should be set up with the hostname or FQDN and returned through `info` endpoint.

> Say you're running in a multi-nic of container environment, what would this be set to?

Not sure but, in k8s, users should specify the value as `my-svc.my-namespace.svc.cluster.local`, I think.

Reviewer: For K8S, if it's in the same namespace you should only need the service name: you only need the namespace if it is in another namespace. The pod name is HOSTNAME too I think.

Author: I see. I tested with ${HOSTNAME} as a default value for this parameter. Then, fluent-bit handles to return actual hostname correctly. I didn't know that the translation mechanism in config_map for environment variable.
Should we change the default value of the hostname parameter?

Reviewer: Not sure, I think it needs to be explicitly a node in the cluster right? We can do if you want, it should be no worse than localhost.

Author: > Not sure, I think it needs to be explicitly a node in the cluster right? 

Yes, it is. Currently, we need to specify the cluster FQDN if needed explicitly.
For now, I wanted to leave as-is. Because localhost is always existing value of FQDN. 
Also, in Windows, there is not always mDNS support. Specifying "localhost" is always safe for every plaftorms.

Reviewer: OK, can you clarify the node discovery aspects then and add that URL?

Author: I didn't find the actual documentation for describing sniffing.
Instead, I found the blog post which describes its feature: https://www.elastic.co/jp/blog/elasticsearch-sniffing-best-practices-what-when-why-how
Should we use it instead of the previous URL?

The plugin supports the following configuration parameters:

| Key | Description | Default value |
| :--- | :--- | :--- |
| buffer\_max\_size | Set the maximum size of buffer. | 4M |
| buffer\_chunk\_size | Set the buffer chunk size. | 512K |
| tag\_key | Specify a key name for extracting as a tag. | `NULL` |
| meta\_key | Specify a key name for meta information. | "@meta" |

> | hostname | Specify hostname or FQDN. This parameter is effective for sniffing node information. | "localhost" |

Reviewer: ```suggestion
| hostname | Specify hostname or FQDN. This parameter can be used for "sniffing" (auto-discovery of) cluster node information. | "localhost" |
```

| version  | Specify Elasticsearch server version. This parameter is effective for checking a version of Elasticsearch/OpenSearch server version. | "8.0.0" |

> **Note:** Elasticsearch cluster uses sniffing to optimize the connections between its cluster and clients. Elasticsearch can build its cluster and dynamically generating for connection list is called as sniffing. `hostname` will be reflected for sniffing information which is handled by the sniffing endpoint.

Reviewer: ```suggestion
**Note:** The Elasticsearch cluster uses "sniffing" to optimize the connections between its cluster and clients.
Elasticsearch can build its cluster and dynamically generate a connection list which is called "sniffing".
The `hostname` will be used for sniffing information and this is handled by the sniffing endpoint.
```
